go/array_string/permutation/permutation.go

```diff
@@ -0,0 +1,29 @@
+// Package permutation has the implementation of operations
+// that has to do do it with strings and permutations
+package permutation
+
+func checkPermutation(inputA, inputB string) bool {
+	// validation
+	if len(inputA) != len(inputB) {
+		return false
+	}
+
+	inputACharSet := make(map[rune]int)
+
+	for _, char := range inputA {
+		inputACharSet[char]++
+	}
+
+	for _, char := range inputB {
+
+		if _, ok := inputACharSet[char]; ok {
+			if inputACharSet[char] == 0 {
+				return false
+			}
+			inputACharSet[char]--
+		} else {
+			return false
+		}
```

Comment:
Another way you can do this is by making use of the fact that maps return the zero value:

```
if inputACharSet[char] == 0 {
  return false
}
inputACharSet[char]--
```

There's no need to change yours to this; just showing you as another way because it's quite specific to Go.

Reply:
I like this approach, thank you.